src/functionHandlers/notarisePdt/v2/handler.ts

```diff
@@ -110,88 +114,79 @@ export const main: Handler = async (
-        body: `${e.title}, ${e.messageBody}`,
-      };
-    }
+    errorWithRef(
+      `Error while validating certificate: ${
+        e instanceof DetailedCodedError ? `${e.title}, ${e.messageBody}` : e
+      }`
+    );
+    return {
+      statusCode: 400,
+      headers: {
+        "x-trace-id": reference,
+      },
+      body:
+        e instanceof DetailedCodedError
+          ? `${e.title}, ${e.messageBody}`
+          : String(e),
+    };
   }
 
-  let result: NotarisationResult | undefined;
-
+  /* 2. Endorsement */
+  let result: NotarisationResult;
+  let directUrl: string;
   try {
-    result = await notarisePdt(
+    ({ result, directUrl } = await notarisePdt(
       reference,
       certificate,
       parseFhirBundle as Bundle,
       testData as TestData[]
-    );
+    ));
   } catch (e) {
-    if (e instanceof Error) {
-      errorWithRef(`Unhandled error: ${e.message}`);
-      return {
-        statusCode: 500,
-        headers: {
-          "x-trace-id": reference,
-        },
-        body: "",
-      };
-    }
+    errorWithRef(`Unhandled error: ${e instanceof Error ? e.message : e}`);
+    return {
+      statusCode: 500,
+      headers: {
+        "x-trace-id": reference,
+      },
+      body: "",
+    };
   }
 
-  /* Notify recipient via SPM (only if enabled) */
+  /* Send SPM notification to recipient (Only if enabled) */
   if (config.notification.enabled) {
     try {
-      if (result && parseFhirBundle && testData && data) {
-        const testType =
-          testData[0].swabTypeCode === config.swabTestTypes.PCR
-            ? "PCR"
-            : testData[0].swabTypeCode === config.swabTestTypes.ART
-            ? "ART"
-            : null;
-        if (
-          config.healthCertNotification.enabled &&
-          testType &&
-          result.directUrl
-        ) {
-          await notifyHealthCert({
-            uin: parseFhirBundle.patient?.nricFin || "",
-            version: "2.0",
-            type: testType,
-            url: result.directUrl,
-            expiry: result.ttl,
-          });
-          delete result.directUrl;
-        } else {
-          await notifyPdt({
-            url: result.url,
-            nric: parseFhirBundle.patient?.nricFin,
-            passportNumber: parseFhirBundle.patient?.passportNumber,
-            testData,
-            validFrom: data.validFrom,
-          });
-        }
-      }
+      await notifyPdt({
+        url: result.url,
+        nric: parseFhirBundle.patient?.nricFin,
+        passportNumber: parseFhirBundle.patient?.passportNumber,
+        testData,
+        validFrom: data.validFrom,
+      });
     } catch (e) {
-      if (e instanceof Error) {
-        errorWithRef(`Notification error: ${e.message}`);
-      }
+      errorWithRef(
+        `SPM notification error: ${e instanceof Error ? e.message : e}`
+      );
+    }
+  }
+
+  /* [NEW] Send HealthCert to SPM wallet (Only if enabled) */
+  if (config.healthCertNotification.enabled) {
```

Comment:
New SPM wallet is only support PCR and ART type for now. That why use the old spm notification as fallback in previous code.